sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosPatch.java

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.patch.PatchOperation;
+import com.azure.cosmos.implementation.patch.PatchOperationCore;
+import com.azure.cosmos.implementation.patch.PatchOperationType;
+import com.azure.cosmos.util.Beta;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Grammar based on RFC: https://tools.ietf.org/html/rfc6902#section-4.1
+ *
+ * Contains a list of Patch operations to be applied on an item. It is applied in an atomic manner and the operations
+ * grammar follows above RFC.
+ *
+ * This can be executed in 3 ways:
+ *  1. Passing this to container in container.patchItem() which requires the id of the item to be patched, partition
+ *      key, the CosmosPatch instance, any CosmosItemRequestOptions and the class type for which response will be parsed.
+ *  2. Add CosmosPatch instance in TransactionalBatch using batch.patchItemOperation() which requires the id of the item
+ *      to be patched, cosmos patch instance and TransactionalBatchItemRequestOptions(if-any) and follow remaining
+ *      steps for batch for it's execution.
+ *  3. Create a bulk item using BulkOperations.getPatchItemOperation which requires the id of the item to be patched,
+ *      cosmos patch instance, partition key and BulkItemRequestOptions(if-any) and follow remaining steps to
+ *      execute bulk operations.
+ */
+@Beta(Beta.SinceVersion.V4_9_0)
+public final class CosmosPatch {
```

Comment:
The usual naming discussion :-)

To me this object represents a Builder/Factory - so CosmosPatchOperationBuilder or CosmosPatchOperationFactory would be more intuitive. But like always - I am not a native speaker maybe patch is not just used as a verb but substantive/noun as well? To me (biased by my Germany linguistic influence a "patch" as a noun would be something like a band-aid. In IT a patch could be a bug fix etc. But it always has the connotation of "healing" - so using something along the lines of PatchOperation would be more intuitive for me.

Reply:
I am not sure. Since we directly say we support patch operations and HTTP uses it in the same context, so like CRUD operations, patch can be directly considered a verb here.

Wiki: In computing, the PATCH method is a request method supported by the Hypertext Transfer Protocol (HTTP) protocol for making partial changes to an existing resource. - https://en.wikipedia.org/wiki/Patch_verb

So may be operation is implicit here? as HTTP considers it directly as a method?

What do you think @moderakh @milismsft ?

Reply:
@rakkuma, I agree with Fabian; lets try to add a suffix to the existing name to make it more specific to our case and avoid any confusion that this might represent an RP action on the respective Cosmos account rather than a document. We can start with "Operation/Operations/Request/Requests" or similar.

Reply:
Cool. Have renamed it to CosmosPatchOperations.